### PR TITLE
fix: Invalid typeclass instances for Cats

### PR DESCRIPTION
**Pull request by Iltotore**
Remove invalid Cats instances for refined types as discussed in #94.

@gvolpe did I miss something or/and removed an actually valid instance? Do you think I should add CommutativeGroup for Positive/Negative types [like in Refined](https://github.com/fthomas/refined/blob/master/modules/cats/shared/src/main/scala/eu/timepit/refined/cats/package.scala#L105)?

### CODE REVIEW

**Comment by gvolpe** [comment]
I would love to see instances for `Semigroup`, `Monoid`, and `CommutativeMonoid` if possible. These are the most common instances used out in the wild. If not in the default `cats` package, I think a `cats.alley` package or so will do.

I can help with testing the lawfulness of the instances if you need help with that part, but I don't know exactly how such instances should be implemented with all the necessary type-level magic 😄

**Comment by Iltotore** [comment]
> I would love to see instances for `Semigroup`, `Monoid`, and `CommutativeMonoid` if possible

I guess the same instances as in Refined? If yes I think it is okay.

**Comment by gvolpe** [comment]
> I guess you mean the same instances as in Refined? If yes I think it is okay.

Yes. There are a few for `Monoid` (which implies `Semigroup`), and more for `CommutativeMonoid`.

**Comment by Iltotore** [comment]
I think I didn't forgot anything

**Review by gvolpe — COMMENTED** [review]
Looks alright besides the minor comments I left 👍🏽

File: cats/src/io/github/iltotore/iron/cats.scala [hunk 1]
```diff
@@ -72,25 +61,54 @@ object cats extends IronCatsInstances:
...
+ * Cats' instances for Iron that need to have a lower priority to avoid ambiguous implicits.
+ */
 private trait IronCatsLowPriority:
-  inline given [A, B](using inline ev: Hash[A]): Hash[A :| B] = ev.asInstanceOf[Hash[A :| B]]
+  inline given [A, C](using inline ev: Hash[A]): Hash[A :| C] = ev.asInstanceOf[Hash[A :| C]]
+
+/**
+ * Arbitrary yet useful instances for some refined types.
+ */
+private trait IronCatsAlleyInstances:
```
[the remark this answers is not in the document]
Comment (Iltotore) [reply cats/src/io/github/iltotore/iron/cats.scala:87]: Do you have an idea for a better name?

File: cats/src/io/github/iltotore/iron/cats.scala [hunk 2]
```diff
@@ -72,25 +61,54 @@ object cats extends IronCatsInstances:
...
+    new CommutativeMonoid[A :| C]:
+
+      override def empty: A :| C = ev.empty.asInstanceOf[A :| C]
+
+      override def combine(a: A :| C, b: A :| C): A :| C = shift.shift(ev.combine(a, b)).asInstanceOf[A :| C]
+
+  inline given negIntCommutativeMonoid[C](using C ==> Negative): CommutativeMonoid[Int :| C] = negMonoid
+  inline given negLongCommutativeMonoid[C](using C ==> Negative): CommutativeMonoid[Long :| C] = negMonoid
+  inline given negFloatCommutativeMonoid[C](using C ==> Negative): CommutativeMonoid[Float :| C] = negMonoid
+  inline given negDoubleCommutativeMonoid[C](using C ==> Negative): CommutativeMonoid[Double :| C] = negMonoid
```
[the remark this answers is not in the document]
Comment (Iltotore) [reply cats/src/io/github/iltotore/iron/cats.scala:114]: I used the same line spacing as you did for the instances (if we're talking about the same thing) but yes I should run ScalaFMT.

[the remark this answers is not in the document]
Comment (gvolpe) [reply cats/src/io/github/iltotore/iron/cats.scala]: No, I'm talking about newlines at end of files. They are missing from `Negshift` and this file, which makes them non-files under POSIX (Unix, Linux, Mac, etc). See the "No newline at end of file" message when you hover over ⛔ 

You should configure your IDE to add newlines on save, not sure if this is something scalafmt can or should do.

[the remark this answers is not in the document]
Comment (gvolpe) [reply cats/src/io/github/iltotore/iron/cats.scala]: Ok, looks like I was looking at an outdated version of this PR. Seems all good now 👍🏽

File: cats/src/io/github/iltotore/iron/cats.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (gvolpe) [reply cats/src/io/github/iltotore/iron/cats.scala:92]: > I would like to separate them for the reasons explained in the discussions

If that's the case, shouldn't they be on a different package such as `cats.alley.*` then?

[the remark this answers is not in the document]
Comment (Iltotore) [reply cats/src/io/github/iltotore/iron/cats.scala:92]: This seems a bit overkill to me. I will follow your advice and put it in IronCatsInstances. We'll see if it causes issues and if it does then we should be able to safely refactor.

**Pull request merged**